cloudferrylib/os/actions/check_filter.py

```diff
@@ -14,69 +15,124 @@
 
 
 from glanceclient import exc as glance_exc
+from cinderclient import exceptions as cinder_exc
 from keystoneclient import exceptions as keystone_exc
 from novaclient import exceptions as nova_exc
 from cloudferrylib.base.action import action
+from cloudferrylib.os.storage import filters as cinder_filters
 from cloudferrylib.utils import utils as utl
 
+import datetime
+
 LOG = utl.get_log(__name__)
 
 
 class CheckFilter(action.Action):
+
+    """CheckFilter class."""
+
     def run(self, **kwargs):
-        """Check filter file and make sure all entries are present in source
-        cloud.
+        """
+        Check filter file.
+
+        Check and make sure all entries are present in source cloud.
+
         """
         search_opts = kwargs.get('search_opts', {})
-        search_opts_img = kwargs.get('search_opts_img', {})
-        search_opts_tenant = kwargs.get('search_opts_tenant', {})
+        self._check_opts_img(kwargs.get('search_opts_img', {}))
+        self._check_opts_vol(kwargs.get('search_opts_vol', {}))
+        self._check_opts_vol_date(kwargs.get('search_opts_vol', {}))
+        self._check_opts_tenant(kwargs.get('search_opts_tenant', {}))
+
+        compute_resource = self.cloud.resources[utl.COMPUTE_RESOURCE]
+        if search_opts and search_opts.get('id'):
+            instances = search_opts['id']
+            for instance_id in instances:
+                LOG.debug('Filtered instance id: %s', instance_id)
+                try:
+                    instance = \
+                        compute_resource.nova_client.servers.get(instance_id)
+                    if instance:
+                        LOG.debug('Filter config check: Instance ID %s is OK',
+                                  instance_id)
+                except nova_exc.NotFound as e:
+                    LOG.error('Filter config check: Instance ID %s '
+                              'is not present in source cloud, '
+                              'please update your filter config. Aborting.',
+                              instance_id)
+                    raise e
```

Comment:
This will fail with the first invalid record in filter file, instead let's have all the invalid IDs listed with just one message, without having user re-run the whole migration multiple times.